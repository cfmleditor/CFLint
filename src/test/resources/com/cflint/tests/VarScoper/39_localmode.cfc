<cfscript>


    function localModeFunction() LocalMode="modern" {
        // This variable is local to the function
        localVar = "I am a local variable";
        return localVar;
    }
    function localModeFunction1() LocalMode="true" {
        // This variable is local to the function
        localVar = "I am a local variable";
        return localVar;
    }
    function localModeFunction2() LocalMode="#true#" {
        // This variable is local to the function
        localVar = "I am a local variable";
        return localVar;
    }
    function localModeFunction3() LocalMode=false {
        // This variable is local to the function
        var localVar = "I am a local variable";
        return localVar;
    }
    function localModeFunction4() LocalMode=false {
        // This variable is local to the function
        localVar = "I am a local variable";
        return localVar;
    }
</cfscript>