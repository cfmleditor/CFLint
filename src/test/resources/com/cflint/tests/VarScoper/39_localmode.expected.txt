{
  "version" : "",
  "timestamp" : 1748956517,
  "issues" : [ {
    "severity" : "ERROR",
    "id" : "MISSING_VAR",
    "message" : "MISSING_VAR",
    "category" : "CFLINT",
    "abbrev" : "MV",
    "locations" : [ {
      "file" : "src/test/resources/com/cflint/tests/VarScoper/39_localmode.cfc",
      "fileName" : "39_localmode.cfc",
      "function" : "localModeFunction4",
      "offset" : 841,
      "column" : 9,
      "line" : 26,
      "message" : "Variable localVar is not declared with a var statement.",
      "variable" : "localVar",
      "expression" : "localVar"
    } ]
  } ],
  "counts" : {
    "totalFiles" : 0,
    "totalLines" : 0,
    "countByCode" : [ {
      "code" : "MISSING_VAR",
      "count" : 1
    } ],
    "countBySeverity" : [ {
      "severity" : "ERROR",
      "count" : 1
    } ]
  }
}